### PR TITLE
Support to record the simulation to a usd file and soft body demos

### PR DESCRIPTION
**Pull request by Jax922**
1. Add USD util class and unit tests for it
2. Add soft body demos 
3. Add USD test to the github workflow actions

### CODE REVIEW

File: python/soft_body_app/soft_application.py [hunk 1]
```diff
@@ -0,0 +1,230 @@
+import os
+import sys
+import numpy as np
+from tqdm import tqdm
```
Comment (erleben) [inline python/soft_body_app/soft_application.py:4]: this package may not be a generally known one. Might be good to add a python inline comment saying this makes an ascii progress bar visualization of a for loop and is helpful for visual feedback?

File: python/soft_body_app/soft_application.py [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+        # A nested loop for simulation and rendering, 
+        # outer loop for rendering, inner loop for simulation computation.
+        for step in tqdm(range(render_steps), desc="Rendering Loop:"):
+            dT = render_dt 
+            while dT > 0:
+                dt = min(dT, sim_dt) 
+                API.simulate(self.scene, dt, debug_on=True)
+                dT -= dt
+                if update_viewer:
+                    self.upadte_viewer(non_update_bodies)
```
Comment (erleben) [inline python/soft_body_app/soft_application.py:223]: Why is the viewer updated more frequently than the render fps?

**Review by erleben — APPROVED** [review]

File: .github/workflows/python-package-conda.yml
[the remark this answers is not in the document]
Comment (Jax922) [reply .github/workflows/python-package-conda.yml:1]: There can not find the `usd-core` package in conda by this command `conda search usd-core` .

File: python/soft_body_app/soft_application.py (hunk 1 above)
Comment (Jax922) [reply python/soft_body_app/soft_application.py:4]: I will submit another pull/request to fix this problem.

File: python/soft_body_app/soft_application.py (hunk 2 above)
Comment (Jax922) [reply python/soft_body_app/soft_application.py:223]: The `update_viewer` function is used to render the simulation processing here. For this purpose, the `update_viewer` should be executed more frequently than the render fps. In other words, the `update_viewer` is real-time to watch the full simulation processing locally.

Comment (erleben) [reply python/soft_body_app/soft_application.py:223]: I find the wording a little confusing, the viewer is actually doing rendering to show the simulation, and saving to a USD file is not really doing any rendering, it is just "baking" the simulation into a file. I guess you can talk about interactive rendering versus offline rendering, but it is still not obvious from the code that there are two types of rendering going on. Perhaps add some inline comments explaining that the viewer is for interactive rendering and tries to keep up with the simulator time-step while rendering is for offline creation of movies and fps control is for the movie creation process?

Comment (Jax922) [reply python/soft_body_app/soft_application.py:223]: Yeah, no problem. I will fix it.

**Pull request merged**